network-api/networkapi/mozfest/factory.py

```diff
@@ -0,0 +1,106 @@
+Faker.add_provider(StreamfieldProvider)
+
+is_review_app = False
+if settings.HEROKU_APP_NAME:
+    is_review_app = True
+
+
+class MozfestPrimaryPageFactory(PageFactory):
+    class Meta:
+        model = MozfestPrimaryPage
+        exclude = ('header_text')
+
+    header = LazyAttribute(lambda o: o.header_text.rstrip('.'))
+    banner = SubFactory(ImageFactory)
+    intro = Faker('paragraph', nb_sentences=3, variable_nb_sentences=False)
+    body = Faker('streamfield', fields=streamfield_fields)
+    header_text = Faker('sentence', nb_words=6, variable_nb_words=True)
+
+
+class MozfestHomepageFactory(MozfestPrimaryPageFactory):
+    class Meta:
+        model = MozfestHomepage
+        exclude = (
+            'header_text',
+            'banner_heading_text'
+        )
+
+    banner_heading = 'Come with an idea, leave with a community.'
+    banner_guide_text = ('Now in its 10th year, the Mozilla Festival is a seven-day '
+                         'gathering of educators, activists, technologists, artists, and '
+                         'young people dedicated to creating a better, healthier open internet.')
+    banner_video_url = Faker('url')
+    prefooter_text = Faker('paragraph', nb_sentences=5, variable_nb_sentences=True)
+    banner_heading_text = Faker('sentence', nb_words=6, variable_nb_words=True)
+
+
+def generate(seed):
+    reseed(seed)
+
+    print('Generating Mozfest Homepage')
+    try:
+        home_page = MozfestHomepage.objects.get(title='Mozilla Festival')
+        print('Homepage already exists')
+    except MozfestHomepage.DoesNotExist:
+        print('Generating a Homepage')
+        # If this is a review app, make the root of the mozfest site a child of the main site
+        if is_review_app:
+            site_root = WagtailPage.objects.get(title='Homepage')
+        else:
+            site_root = WagtailPage.objects.get(id=1)
+
+        home_page = MozfestHomepageFactory.create(
+            parent=site_root,
+            title='Mozilla Festival',
+            slug=None
+        )
+
+    reseed(seed)
+
+    if not is_review_app:
+        print('Creating MozFest Site record in Wagtail')
+        tds = settings.TARGET_DOMAINS
+        if tds and len(tds) > 1:
+            # Assume that tds[0] is the main mofo domain, and tds[1] is the Mozfest domain
+            hostname = tds[1]
+            port = 80
+        else:
+            # use a localhost domain (must be set in /etc/hosts)
+            hostname = 'mozfest.localhost'
+            port = 8000
+
+        WagtailSite.objects.create(
+            hostname=hostname,
+            port=port,
+            root_page=home_page,
+            site_name='Mozilla Festival',
+            is_default_site=False
+        )
+
+    print('Generating Mozfest sub-pages')
+    [MozfestPrimaryPageFactory.create(
+        parent=home_page,
+        title=title
+    ) for title in ['About', 'Spaces', 'Tickets', 'Team', 'Sponsors']]
```

Comment:
there is no need for an `about` page anymore (it's been removed)

Reply:
made the change!